packages/ui/src/ui/pages/query-tracker/FileEditor/FileEditor.tsx

```diff
@@ -0,0 +1,96 @@
+import React, {FC, useRef} from 'react';
+import {createSelector} from 'reselect';
+import {
+    selectDeletedFiles,
+    selectFileEditor,
+    setFileEditor,
+} from '../module/queryFilesForm/queryFilesFormSlice';
+import {changeQueryFile} from '../module/queryFilesForm/actions';
+import {getQueryFiles} from '../module/query/selectors';
+import {useDispatch, useSelector} from 'react-redux';
+import FileIcon from '@gravity-ui/icons/svgs/file.svg';
+import XmarkIcon from '@gravity-ui/icons/svgs/xmark.svg';
+import SquareIcon from '@gravity-ui/icons/svgs/square.svg';
+import LayoutColumnsIcon from '@gravity-ui/icons/svgs/layout-columns.svg';
+import {Button, Icon, Text} from '@gravity-ui/uikit';
+import './FileEditor.scss';
+import cn from 'bem-cn-lite';
+import MonacoEditor, {MonacoEditorConfig} from '../../../components/MonacoEditor';
+import * as monaco from 'monaco-editor';
+import {QueryFile} from '../module/api';
+
+const block = cn('file-editor');
+
+const MONACO_CONFIG: MonacoEditorConfig = {
+    fontSize: 14,
+    language: 'plaintext',
+    renderWhitespace: 'boundary',
+    minimap: {
+        enabled: true,
+    },
+};
+
+const selector = createSelector(
```

Comment:
It is better to follow common approaches and move the selector to a separate file near the definition of the corresponding reducer.